Butterbear

Sneeker134 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

As my second PR, I have created an abomination; the BUTTER BEAR!
![BUTTERBEAR](https://user-images.githubusercontent.com/53362182/65025389-3a1f5180-d8fc-11e9-9e4e-57e3e2a09210.JPG)

The butter bear is really similar to Cak; it is a craftable organism that gives whoever's brain was used to make it control of the body. It is constructed using a brain, a heart, a whopping 10 sticks of butter, 5 slabs of meat, 50 blood, and one teslium. The crew can eat the bear by smacking it, but it has a large regeneration factor to counteract this. When the bear slaps people, it has a significant chance to slip its unfortunate victim.
## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Its just a fun thing to do. I play quite a bit of chef, and on a slow shift one of my favorite things to do is to try and convince an assistant to let me put their brain inside a cake cat body and then watch them run around the station as everyone tries to eat them. The butter bear gives you another option to do something like this if you really want it, though it pains me to admit, lacks the pun.

## Changelog
:cl:
add: Added new craftable Mob, the Butter Bear.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sneeker134 committed:
Created the almighty BUTTERBEAR!!!

Sneeker134 committed:
Fixed a subcategory issue.

Sneeker134 committed:
Added sound effect to slip, heavily nerfed duration of knockdown

Sneeker134 committed:
Merge branch 'master' into BUTTERBEAR

81Denton commented on code/modules/mob/living/simple_animal/hostile/bear.dm:
```diff
@@ -99,8 +99,57 @@
 		qdel(src)
 
+mob/living/simple_animal/hostile/bear/butter //The mighty companion to Cak. Several functions used from it.
+	name = "Kerrygold"
```
Change the name to something that's not exactly the trademark name please

MrDoomBringer replied:
```suggestion
	name = "Terrygold"
```

Sneeker134 replied:
Good suggestion MrDoomBring! I'll merge it.

Sneeker134 committed:
Changed default name of mob

Trademarked name I guess was a bad idea, but the suggested name is pretty good!

Co-Authored-By: MrDoomBringer <emmanuelssr@gmail.com>

OverlyGeneric commented:
Technically, the name is already a pun.

It doesn't fit thematically but it seems like a portmanteau for "butterbean" which are lima beans.

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/bear.dm:
```diff
@@ -99,8 +99,57 @@
+	faction = list("neutral")
+	obj_damage = 0
+	melee_damage_lower = 5
+	melee_damage_upper = 10
```
You, uh, MIGHT not want to give a creature with 60 health and the ability to regenerate 10 HP per tick and the ability to knock people down on a hit the ability to directly hurt people. Make its melee_damage_lower and melee_damage_upper stats 0 and buff its chance to knock people down on each hit to, say, 100% (but make it unable to "slip" prone people), though, and this should be fine.

ATH1909 replied:
Also, keep in mind that butterbears will be able to wear bear armor like normal bears can (which I think should be left in as a feature) to gain incredible stat buffs:
```
		A.armored = TRUE
		A.maxHealth += 60
		A.health += 60
		A.armour_penetration += 20
		A.melee_damage_lower += 5
		A.melee_damage_upper += 5
```

Sneeker134 replied:
I think all your suggestions are pretty good ones; frankly I was just sort of designing something to be goofy and never gave it much in the way of balance implications. I dropped the melee damage down to 1, attacks by the butterbear now slip targets 100% of the time but don't affect prone targets, gave it an object damage of 11 so it doesn't get trapped, and threw the russian faction in there.

I think the bear armor thing is pretty funny, so I think it deserves to stay in. I intended them to more of just be annoying and not really much of combatants, but if someone wants to go through all that work to try and give them some power I think thats fair.

Sneeker134 committed:
Balance changes

Bear attack damaged nerfed down to 1. Now is part of the russian faction. Slips now work 100% of the time, but you are unable to slip targets that are already prone.

Sneeker134 committed:
Had some indentation errors; oops!

Fox-McCloud commented on code/modules/mob/living/simple_animal/hostile/bear.dm:
```diff
@@ -99,8 +99,58 @@
+	if(isliving(target))
+		var/mob/living/L = target
+		if(!(L.mobility_flags & MOBILITY_STAND))
+		else
```
why not just.... `if(L.mobility_flags & MOBILITY_STAND))` instead?

Sneeker134 replied:
Uh, thats a good question. I asked how to tell if a mob is prone in the discord, and thats what I was given so I just used it; didn't give it too much thought. It _should_ work so I don't think its that big of an issue.

Fox-McCloud replied:
This is really gross formatting though, and unless you have a good reason to have an empty `if` statement, it'd be more preferable for have cleaner, more easily readable code.

Sneeker134 replied:
Fair enough. I've changed it to just a check to see if they're standing, rather than an if-else to see if they aren't standing.

cacogen commented:
we need more hp references in-game

81Denton commented:
Ok the new name is amazing, thank you for that pun

Sneeker134 committed:
Less icky

Removed If Else loop for just an if

Difrit commented:
![image](https://user-images.githubusercontent.com/12451752/65222318-b362a400-dab6-11e9-9f73-7a2c593d571c.png)

IndieanaJones commented:
Seems pretty strong.  This combined with bear armor and a fugu could prove to be very formidable.

Sneeker134 commented:
> Seems pretty strong. This combined with bear armor and a fugu could prove to be very formidable.

If it proves to be too strong/I'm told to nerf it or no merge, I think removing the fugu interaction is probably the route to do it. I think adding the bear armor is something thats kind of cool, and I'd like to see if it works while having the fugu interaction. If its something people think people are going to want to tunnel into for combat stats and such, I don't think thats what the Butter Bear is really designed to be, and changes will be made accordingly. If its something that happens to be strong but not overBEARing, hard to pull off, and pretty rare, then I think thats fine.

MrDoomBringer commented on code/modules/mob/living/simple_animal/hostile/bear.dm:
```diff
@@ -99,8 +99,57 @@
+	to_chat(src, "<span class='big bold'>You are a butter bear!</span><b> You're a mostly harmless bear/butter hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
+	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
+	free butter to the station!</b>")
+	var/new_name = stripped_input(src, "Enter your name, or press \"Cancel\" to stick with Kerrygold.", "Name Change")
```
```suggestion
	var/new_name = stripped_input(src, "Enter your name, or press \"Cancel\" to stick with Terrygold.", "Name Change")
```

Sneeker134 committed:
Update old name

Co-Authored-By: MrDoomBringer <emmanuelssr@gmail.com>
